### ISSUE TITLE
src/dns.c: error handling issue suspected by coverity (new finding)

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#1129]
```
*** CID 1446652:  Error handling issues  (CHECKED_RETURN)
/src/dns.c: 72 in dns_connect_nameserver()
66                               ns->counters->id, ns->id);
67                      close(fd);
68                      return -1;
69              }
70     
71              /* Make the socket non blocking */
>>>     CID 1446652:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "fcntl(fd, 4, 2048)" without checking return value. This library function may fail and return an error code.
72              fcntl(fd, F_SETFL, O_NONBLOCK);
73     
74              /* Add the fd in the fd list and update its parameters */
75              dgram->t.sock.fd = fd;
76              fd_insert(fd, dgram, dgram_fd_handler, MAX_THREADS_MASK);
77              fd_want_recv(fd);
```

**Comment by EmericBr** [comment]
fcntl returns is never checked in the code, should be a false positive. Or may be fixed byr proposed patch  #1127 if coverity suspects fd could be -1.

**Comment by chipitsine** [comment]
I marked as false positive
